ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/validation/BlockValidatorTest.java

```diff
@@ -286,7 +296,31 @@ void shouldReturnInvalidOnWrongExecutionPayloadTimestamp(SpecContext specContext
                         specContext.getDataStructureUtil().randomExecutionPayload()));
 
     InternalValidationResult result =
-        blockValidator.validate(signedBlockAndState.getBlock()).join();
+        blockValidator.validate(signedBlockAndState.getBlock(), Optional.empty()).join();
+    assertTrue(result.isReject());
+  }
+
+  @TestTemplate
+  void shouldReturnInvalidForInconsistentBlockAndBlobsSidecar(final SpecContext specContext) {
+    specContext.assumeEip4844Active();
+
+    storageSystem = InMemoryStorageSystemBuilder.buildDefault(spec);
+    storageSystem
+        .chainUpdater()
+        .initializeGenesisWithPayload(
+            false, specContext.getDataStructureUtil().randomExecutionPayloadHeader());
+    recentChainData = storageSystem.recentChainData();
+    blockValidator = new BlockValidator(spec, recentChainData);
+
+    final UInt64 nextSlot = recentChainData.getHeadSlot().plus(ONE);
+    storageSystem.chainUpdater().setCurrentSlot(nextSlot);
+
+    SignedBeaconBlock block = storageSystem.chainBuilder().generateBlockAtSlot(nextSlot).getBlock();
+
+    final BlobsSidecar blobsSidecar = specContext.getDataStructureUtil().randomBlobsSidecar();
+
+    InternalValidationResult result =
+        blockValidator.validate(block, Optional.of(blobsSidecar)).join();
     assertTrue(result.isReject());
```

Comment:
I'd separate clarification of both implemented cases (only bad slot and only bad root) and also add valid test

Reply:
I agree, I was a bit lazy here because I think these consistency checks will go away.. adding missing tests

Reply:
added